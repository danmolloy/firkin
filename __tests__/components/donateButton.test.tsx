import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import DonateButton from "@/components/donateButton"

describe("<DonateButton />", () => {
  beforeEach(() => {
    render(<DonateButton />)
  })
  it("button is in the document with text and attrs type and role", () => {
    const btn = screen.getByText("Donate")
    expect(btn).toBeInTheDocument()
    expect(btn).toHaveAttribute("type", "submit")
    expect(btn).toHaveAttribute("role","link")

  })
  it("form has attrs action and post with expected args", () => {
    const form = screen.getByTestId("donate-btn")
    expect(form).toBeInTheDocument()
    expect(form).toHaveAttribute("action", "/donate/api")
    expect(form).toHaveAttribute("method","POST")
  })
  it("matches snapshot", () => {
    const form = screen.getByTestId("donate-btn")
    expect(form).toMatchSnapshot()
  })
})