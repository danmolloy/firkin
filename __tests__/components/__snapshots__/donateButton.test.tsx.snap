// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DonateButton /> matches snapshot 1`] = `
<form
  action="/donate/api"
  data-testid="donate-btn"
  method="POST"
>
  <button
    class="hover:text-yellow-400 rounded font-bold text-lg font-text"
    role="link"
    type="submit"
  >
    Donate
  </button>
</form>
`;
